### PR TITLE
fixed submitting form with amount blank and had 0 amount not show on …

### PR DESCRIPTION
**Pull request by hammypwns**
…recipe page and recipes forms

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,6 +55,7 @@
     "react-toast-notifications": "^2.4.0",
     "react-use-auth": "^0.5.1",
     "theme-ui": "^0.2.44",
+    "typescript": "^3.7.5",
```
Comment (garrowp) [inline package.json]: what's this for?

File: src/components/recipe-form.js [hunk 2]
```diff
@@ -131,7 +141,7 @@ const UnitDropdown = props => (
 const RecipeForm = ({
   recipe_id = null,
   name = '',
-  ingredients = [{ name: '', unit: '', amount: '' }],
+  ingredients = [{ name: '', unit: '', amount: 0 }],
```
Comment (garrowp) [inline src/components/recipe-form.js]: why are we defaulting it to 0? Don't we want it to be blank at first?

File: src/utils/createRecipeObject.js [hunk 3]
```diff
@@ -11,14 +11,29 @@ const createRecipeObject = (
...
   const latest_version = version + 1
   const ingredients = {
     data: values.ingredients.map(ingredient => {
       if (ingredient.amount.toString().includes('/')) {
-        const [a, b] = ingredient.amount.toString().split('/')
-        ingredient.amount = Number(a / b)
+        let pre = 0
+        let post = ingredient.amount
+        if (ingredient.amount.toString().includes(' ')) {
+          ;[pre, post] = ingredient.amount.toString().split(' ')
```
Comment (garrowp) [inline src/utils/createRecipeObject.js:32]: There's an extra semicolon

File: package.json (hunk 1 above)
Comment (hammypwns) [reply package.json]: gatsby develop won't run without it

File: src/components/recipe-form.js (hunk 2 above)
Comment (hammypwns) [reply src/components/recipe-form.js]: you gotta point. The result is the same due to some code later, but it should be '' yee

File: src/utils/createRecipeObject.js (hunk 3 above)
Comment (hammypwns) [reply src/utils/createRecipeObject.js:32]: weird. fixed

**Pull request merged**